### PR TITLE
Add Automatic Record Creation code for customizing Email Multi-Related functionallity

### PR DESCRIPTION
**Pull request by barbaricprogrammer**
This code provides customers (Specifically requested by FSI customers) with the ability to customize which cases are related to an email.  It provides a customization that removes any case created from a queue monitored by Automatic Record Creation (ARC) when that queue is no longer in the recipients.  (ex. email sent to support and engineering queues.  Case S created for the email on the support queue and case E for the email on the engineering queue.  The reply email is sent to only the support queue and this customization would remove case E from the related parties on that reply email).

### CODE REVIEW

File: customer-service/automatic-record-creation/RemoveUnrefrencedQueues/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+---
+languages:
+- csharp
+products:
+- dynamics-customer-service
+page_type: sample
+level: beginner
+role: developer
+description: "Sample that shows how Automatic Record Creation for Dynamics 365 Customer Service can be customized to remove unrefrenced queues from recieved emails"
```
Comment (phecke) [inline customer-service/automatic-record-creation/RemoveUnrefrencedQueues/README.md:9]: Spelling error on "unrefrenced" and "recieved"

**Review by JimDaly — CHANGES_REQUESTED** [review]
Please change the name of the project.  `Unreferenced` is misspelled.


RemoveUnrefrencedQueues s/b RemoveUnreferencedQueues

File: customer-service/automatic-record-creation/RemoveUnrefrencedQueues/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+description: "Sample that shows how Automatic Record Creation for Dynamics 365 Customer Service can be customized to remove unreferenced queues from received emails"
+---
+
+# Sample: ARC Multi-Related - Remove Unrefrenced Queues
+
+This sample shows how to write a simple plug-in that updates the activity parties on an email removing any related item's created by ARC that are no longer party to the email.  This assumes that the org has enabled ARC's Multi-Related for Email feature.
+
+## Use cases Covered
+
+1. An external user sends an email to a queue monitored by Automatic Record Creation (ARC).  ARC creates a case for that email and adds the created case to the email's related activity party list. Because of this relationship the email is shown on the case's timeline view.  The user then forwards an email from that chain to another mailbox (for example, the agent's manager).  This custimization allows for organizations that  do not want that forwarded email on the timeline like it would be by default.
```
Comment (JimDaly) [inline customer-service/automatic-record-creation/RemoveUnrefrencedQueues/README.md:18]: custimization s/b customization

File: customer-service/automatic-record-creation/RemoveUnrefrencedQueues/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+2. Similarlly, If a user replies to an email that was sent to multiple ARC Rules (for example, email sent to support and engineering queues) and the user removes one of those mailboxes from a reply, this customization would remove the cases not assosciated with a recipient. For example, if sent to only support queue, any cases created from the engineering queue would be removed.
+
+## How to run this sample
+
+1. Download or clone the [Dynamics 365 Samples](https://github.com/microsoft/Dynamics365-Apps-Samples) repo so that you have a local copy. This sample is located under customer-service\automatic-record-creation\RemoveUnrefrencedQueues.
```
Comment (JimDaly) [inline customer-service/automatic-record-creation/RemoveUnrefrencedQueues/README.md:23]: When you change the project name to `RemoveUnreferencedQueues` make sure you change all references to it.

File: customer-service/automatic-record-creation/RemoveUnrefrencedQueues/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+2. Open the sample solution in Visual Studio, navigate to the project's properties, and verify the assembly will be signed during the build. Press F6 to build the sample's assembly (RemoveUnrefrencedQueues.dll).
+3. Run the Plug-in Registration tool and register the sample's assembly in the D365 server's sandbox and database. When registering a plug-in step, specify the "Create" message, email table, and syncronous mode on the Post Create stage.
```
Comment (JimDaly) [inline customer-service/automatic-record-creation/RemoveUnrefrencedQueues/README.md:25]: syncronous s/b synchronous

File: customer-service/automatic-record-creation/RemoveUnrefrencedQueues/README.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+4. Create two or more Automatic Record Creation rules with unique queue mailboxes.
+5. Trigger case creation of the ARC rule by sending an email to the configured queue for one of the ARC rules.
+6. Test the plugin by replying or forwarding that email to different ARC enabled queues.
+6. When you are done testing, unregister the plugin assembly and step.
```
Comment (JimDaly) [inline customer-service/automatic-record-creation/RemoveUnrefrencedQueues/README.md:29]: In markdown, you don't need to number items. You should just always use `1. `, and the rendering will number it for you.

File: customer-service/automatic-record-creation/RemoveUnrefrencedQueues/README.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+Note that the plugin is run syncronously so that at no point does the email show on the timelines for related items that would be removed.
```
Comment (JimDaly) [inline customer-service/automatic-record-creation/RemoveUnrefrencedQueues/README.md:31]: syncronously s/b synchronously

File: customer-service/automatic-record-creation/RemoveUnrefrencedQueues/README.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+In order to simulate the scenario described in [What this sample does](#what-this-sample-does), the sample will do the following:
+
+- Look at an email.
+- Find the related activity parties on the email.
+- Find the originating queue for any related entities.
+- Remove any related activity parties where the originating queue is no longer party to this email. For example, remove the case created by the support queue if this email no longer has the support queue in the to/cc/bcc fields.
+
+### Demonstrates
+
+1. How to get the relatied parties
```
Comment (JimDaly) [inline customer-service/automatic-record-creation/RemoveUnrefrencedQueues/README.md:48]: relatied s/b related

File: customer-service/automatic-record-creation/RemoveUnrefrencedQueues/README.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+2. How to find the originating queue for the related parties created by ARC
+3. How to update the related parties on the email
+
+### See also
+
+[Write a plug-in](https://docs.microsoft.com/en-us/powerapps/developer/common-data-service/write-plug-in)
```
Comment (JimDaly) [inline customer-service/automatic-record-creation/RemoveUnrefrencedQueues/README.md:54]: Please update these URLs:

`https://docs.microsoft.com/en-us/powerapps/developer/common-data-service/write-plug-in` s/b `https://learn.microsoft.com/power-apps/developer/data-platform/write-plug-in`


`https://docs.microsoft.com/en-us/powerapps/developer/common-data-service/register-plug-in` s/b `https://learn.microsoft.com/power-apps/developer/data-platform/register-plug-in`

**Review by JimDaly — CHANGES_REQUESTED** [review]
Several misspellings and URLs that should be updated.

In addition to the other review that the name of the project should be changed.

File: customer-service/automatic-record-creation/RemoveUnrefrencedQueues/README.md (hunk 4 above)
Comment (barbaricprogrammer) [reply customer-service/automatic-record-creation/RemoveUnrefrencedQueues/README.md:25]: OK i'll install a spellchecker into VS Code :-) Sorry and thanks for the spelling help.

**Pull request merged**